### PR TITLE
Fix overlay single date picker to open to selected date.

### PR DESCRIPTION
**Pull request by cvializ**
Fixes https://github.com/ampproject/amphtml/issues/17936

Also fixes static pickers showing wrong initial month when set from `src` json.

### CODE REVIEW

File: extensions/amp-date-picker/0.1/amp-date-picker.js [hunk 1]
```diff
@@ -476,17 +476,14 @@ export class AmpDatePicker extends AMP.BaseElement {
...
-
       this.state_ = this.getInitialState_();
-      this.render(this.state_);
-      this.setupListeners_();
     });
   }
 
   /** @override */
   layoutCallback() {
+    const srcAttributesP = this.setupSrcAttributes_();
```
Comment (aghassemi) [inline extensions/amp-date-picker/0.1/amp-date-picker.js]: nit: `P` looked like a type until I noticed it is a promise.

File: extensions/amp-date-picker/0.1/single-date-picker.js [hunk 2]
```diff
@@ -82,12 +82,32 @@ function createSingleDatePickerBase() {
...
 }
 
+/**
+ * Fixes bug where overlay single date pickers do not open to
+ * the month containing the selected date.
+ * https://github.com/airbnb/react-dates/issues/931
+ * @param {function(new:React.Component, !Object)} WrappedComponent A date-picker component to wrap
+ * @return {function(new:React.Component, !Object)} A class with a preset focus prop
+ */
+function withFocusTrueHack(WrappedComponent) {
```
Comment (aghassemi) [inline extensions/amp-date-picker/0.1/single-date-picker.js]: wow

**Comment by JakobEichler** [comment]
Hello, since Monday we had no bookings on our mobile version of www.parkscanner.de 

The datepicker was broken, and we made no changes on the datepicker and the AMP site at all.

I do not know if the issues came with this update or with an other one, anyways here are two sites, which worked before any AMP updates.

https://www.parkscanner.de/amp/bug-report2.html
https://www.parkscanner.de/amp/bug-report.html

Now I fixed those datepickers temporarily by removing the display: flex; attribute on the .amp-date-picker-calendar-container

What has not yet been fixed is the way we display conditional messages in the info template depending on which date needs to be entered. This worked some days ago and now also stopped to work. 

Earlier bellow our datepicker on (for example www.parkscanner.de/amp/ ) only one of these messages was displayed: 

Bitte wähle Dein Abreisedatum aus.
Bitte wähle Dein Anreisedatum aus.

Now I see both of them, as the .hide action does not work anymore before the template is rendered fully.

**Comment by cvializ** [comment]
Hi @JakobEichler! Thanks for letting us know about your issue. I created #19109 to help track our investigation and diagnosis. You can check for updates there.

**Pull request merged**